### PR TITLE
Image Cache system implemented

### PR DESCRIPTION
**Pull request by JancoDVTDev**
## Image cache

- Build an image cache that will store photos to the user's device.
- Once the app opens again and the user sign in, it will only download images that was added by another person and not the whole album.

### CODE REVIEW

File: CacheHelper/CacheHelper.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+public class CacheHelper {
+
+    public init() { }
+
+    func saveImage(imageName: String, image: UIImage) {
+        guard let documentsDirectory = FileManager.default.urls(for: .documentDirectory,
+                                                                in: .userDomainMask).first else { return }
+
+        let fileName = imageName
```
Comment (AkuaQ) [inline CacheHelper/CacheHelper.swift]: Please watch out for the spacing in this file.

**Review by AkuaQ — APPROVED** [review]
Apart from that, looks good to me

File: CacheHelper/CacheHelper.swift (hunk 1 above)
Comment (JancoDVTDev) [reply CacheHelper/CacheHelper.swift]: Thanks

File: camPod/Classes/Photos/ViewModel/PhotosViewModel.swift [hunk 2]
```diff
@@ -10,42 +10,87 @@ import Foundation
...
-        if imagePaths.contains("") {
+        for imagePath in imagePaths {
+            cacheHelper?.loadImageFromDisk(fileName: imagePath, { (imagePathToUpdate, cachedImage) in
+                count += 1
+                if let imagePathToUpdate = imagePathToUpdate {
+                    imagePathsToUpdate.append(imagePathToUpdate)
+                } else {
+                    cachedPhotoModels.append(PhotoModel(name: imagePath, image: cachedImage!))
+                }
+                if count == imagePaths.count {
```
Comment (Zamo22) [inline camPod/Classes/Photos/ViewModel/PhotosViewModel.swift:29]: It seems like you're checking here if the loop is finished before the completion. Why not just perform the completion after the for loop ? Unless I'm missing something

Comment (JancoDVTDev) [reply camPod/Classes/Photos/ViewModel/PhotosViewModel.swift:29]: I found that the completion() wil run even it the loop does not finish which then cause the variable to be nil

File: camPod/Classes/CacheHelper/CacheHelper.swift [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    func saveImage(imageName: String, image: UIImage) {
+        guard let documentsDirectory = FileManager.default.urls(for: .documentDirectory,
+                                                                in: .userDomainMask).first else { return }
+
+        let fileName = imageName
+        let fileURL = documentsDirectory.appendingPathComponent(fileName)
+
+        guard let data = image.jpegData(compressionQuality: 1) else { return }
+
+        if (FileManager.default.fileExists(atPath: fileURL.path)) {
```
Comment (Zamo22) [inline camPod/Classes/CacheHelper/CacheHelper.swift:23]: Very minor but there's no need for extra brackets here. You can just have:
`FileManager.default.fileExists(atPath: fileURL.path)`

Comment (JancoDVTDev) [reply camPod/Classes/CacheHelper/CacheHelper.swift:23]: Cool thanks, Codacy also told me hahaha

**Pull request merged**